Source/CarthageKit/Project.swift

```diff
@@ -1521,3 +1559,44 @@ public func cloneOrFetch(
 				}
 		}
 }
+
+/// Invokes otool -L for a given executable URL.
+///
+/// - Parameter executableURL: URL to a valid executable.
+/// - Returns: Array of the Shared Library ID that are linked against given executable (`Alamofire`, `Realm`, etc).
+/// System libraries and dylibs are omited.
+internal func sharedLinkedFrameworks(for executableURL: URL) -> SignalProducer<[String], CarthageError> {
+    return Task("/usr/bin/xcrun", arguments: ["otool", "-L", executableURL.path.spm_shellEscaped()])
+        .launch()
+        .mapError(CarthageError.taskError)
+        .ignoreTaskData()
+        .filterMap { data -> String? in
+            return String(data: data, encoding: .utf8)
+        }
+        .map(sharedLinkedFrameworkIDs(from:))
+}
+
+/// Stripping linked shared frameworks from
+/// @rpath/Alamofire.framework/Alamofire (compatibility version 1.0.0, current version 1.0.0)
+/// to Alamofire as well as filtering out system frameworks and various dylibs.
+/// Static frameworks and libraries won't show up here, so we can ignore them.
+///
+/// - Parameter input: Output of the otool -L
+/// - Returns: Array of Shared Framework IDs.
+internal func sharedLinkedFrameworkIDs(from input: String) -> [String] {
+    guard let regex = try? NSRegularExpression(pattern: "@rpath.*\\.framework\\/([\\w]+)") else {
```

Comment:
@blender we can rely on the `[\\w]+\\.framework\\/([\\w]+)` instead. This will give us also system frameworks like Foundation, however I don't see any performance penalty because of this. Will update regex later today.